src/openrct2/drawing/Drawing.Sprite.cpp

```diff
@@ -282,6 +283,16 @@ bool gfx_load_g2()
         // Read element data
         _g2.data = fs.ReadArray<uint8_t>(_g2.header.total_size);
 
+        if (_g2.header.num_entries != EXPECTED_SPRITES)
+        {
+            log_error("Mismatched g2.dat size");
+            if (!gOpenRCT2Headless)
+            {
+                auto uiContext = GetContext()->GetUiContext();
+                uiContext->ShowMessageBox("Mismatched g2.dat size");
```

Comment:
I feel this could be further embellished perhaps stating that files incorrectly installed.

Reply:
And provide counts received and expected. If possible, also path to g2.dat.